sinon-spy-react.js

```diff
@@ -2,19 +2,55 @@ var React = require('react');
 var ReactDOM = require('react-dom');
 var sinon = require('sinon');
 
-function spyOnReactClass(reactClass, methodName) {
+function spyOnComponentMethod(reactClass, methodName) {
     var classProto = reactClassPrototype(reactClass);
-    var spy = sinon.spy();
+    var spy;
+    var on;
+    var idx;
 
-    if (classProto.__reactAutoBindMap) { // React 0.14.x
-      classProto.__reactAutoBindMap[methodName] = spy;
+    if (classProto.__reactAutoBindMap) { // React 0.14.x1
+      if(typeof classProto.__reactAutoBindMap[methodName] !== 'undefined' ){
+        spy = classProto.__reactAutoBindMap[methodName] = sinon.spy(classProto.__reactAutoBindMap[methodName]);
```

Comment:
@apires One question: What exactly does it when you pass the original function to the spy?